starter_code/src/movies.js

```diff
@@ -1,20 +1,118 @@
 /* eslint no-restricted-globals: 'off' */
-// Turn duration of the movies from hours to minutes 
+// Turn duration of the movies from hours to minutes
+// How to change the value of the same key in all objects of an array
 
+function turnHoursToMinutes(movies) {
+  var time = movies.map(function(oneMovie) {
+    var A = {};
+    var hoursIntoMinutes = Number(oneMovie.duration.charAt(0) * 60);
+    var minutes = Number(
+      oneMovie.duration.charAt(3) + oneMovie.duration.charAt(4)
+    );
+    console.log(hoursIntoMinutes + minutes);
+    return (oneMovie.duration = hoursIntoMinutes + minutes);
+  });
+  return time;
+}
 
-// Get the average of all rates with 2 decimals 
+console.log(turnHoursToMinutes(movies));
 
+// Get the average of all rates with 2 decimals
+
+function ratesAverage(movies) {
+  var average = movies.reduce(function(sum, oneMovie) {
+    return sum + Number(oneMovie.rate);
+  }, 0);
+  return Number((average / movies.length).toFixed(2));
+}
+
+console.log(ratesAverage(movies));
 
 // Get the average of Drama Movies
 
+function dramaMoviesRate(movies) {
+  var dramaMovies = movies.filter(function(oneMovie) {
+    return oneMovie.genre.indexOf("Drama") > -1;
+  });
+  if (dramaMovies.length === 0) {
+    return undefined;
+  }
+  return ratesAverage(dramaMovies);
+}
+
+console.log(dramaMoviesRate(movies));
 
 // Order by time duration, in growing order
+// Will do when Iteration 1 is solved
 
+// Order by date
+//Cannot solve the 2 Jasmine errors :
+// - Return the new array in growing order
+// -If two movies have the same length, order them alphabetically by their title
+
+function orderByDuration(movies) {
+  var orderedYears = movies.sort(function(itemA, itemB) {
+    if (itemA.year < itemB.year) {
+      return -42;
+    } else {
+      return 89;
+    }
+  });
+  return orderedYears;
+}
 
 // How many movies did STEVEN SPIELBERG
 
+function howManyMovies(movies) {
+  if (movies.length === 0) {
+    return undefined;
+  }
+  var moviesSpielberg = movies.filter(function(oneMovie) {
+    return (
+      oneMovie.director === "Steven Spielberg" &&
+      oneMovie.genre.indexOf("Drama") > -1
+    );
+  });
+  return (
+    "Steven Spielberg directed " + moviesSpielberg.length + " drama movies!"
+  );
+}
 
 // Order by title and print the first 20 titles
+//Cannot solve the 2 Jasmine errors :
+// - You should order them alphabetically.
+// - You should return the top20 after order them alphabetically.
+
+function orderAlphabetically(movies) {
```

Comment:
You've made this a bit more complicated than you needed to.

In your first sort you should check for the title not the rate 

You only need to sort once. Once you have a sorted array, you can slice that if it's more than 20 or return the entire array if it's below 20. 

```
function orderAlphabetically(movies) {
  var orderedRate = movies.sort(function(itemA, itemB) {
    if (itemA.title < itemB.title) {
      return -42;
    } else {
      return 89;
    }
  });
  var top20Movies = [];
  var limit = 20;
  if (orderedRate.length < 20) {
    limit = orderedRate.length;
  }
  for (var i = 0; i < limit; i++) {
    top20Movies.push(orderedRate[i].title);
  }

  return top20Movies;
}
```

if this doesn't make sense, let me know and I can explain :)